movies/movies/Service.swift

```diff
@@ -0,0 +1,112 @@
+        value: UserDefaults.standard.string(forKey: Constants.apiValueKeyName)
+    )
+    private let queryItemLanguage = URLQueryItem(
+        name: Constants.queryItemLanguageName,
+        value: Constants.queryItemLanguageValue
+    )
+    private var category = PurchaseEndPoint.popular
+
+    func loadFilms(page: Int, completion: @escaping (Result) -> Void) {
+        loadFilms(page: page, api: category, completion: completion)
+    }
+
+    func loadFilms(page: Int, api: PurchaseEndPoint, completion: @escaping (Result) -> Void) {
+        category = api
+
+        let queryItemPage = URLQueryItem(name: Constants.queryItemPageName, value: "\(page)")
+
+        var components = URLComponents()
+        components.scheme = Constants.componentScheme
+        components.host = Constants.componentsHost
+        components.path = api.rawValue
+        components.queryItems = [queryItemKey, queryItemLanguage, queryItemPage]
+        guard let url = components.url else { return }
+        let task = session.dataTask(with: url) { data, _, _ in
+            guard let data = data else { return }
+            do {
+                let result = try JSONDecoder().decode(Result.self, from: data)
+                completion(result)
+            } catch {
+                print(error)
+            }
+        }
+        task.resume()
+    }
+
+    func loadFilm(index: Int, completion: @escaping (Film) -> Void) {
+        var components = URLComponents()
+        components.scheme = Constants.componentScheme
+        components.host = Constants.componentsHost
+        components.path = Constants.componentsPath + "\(index)"
+        components.queryItems = [queryItemKey, queryItemLanguage]
+
+        guard let url = components.url else { return }
+        print(url)
+        let task = session.dataTask(with: url) { data, _, _ in
+            guard let data = data else { return }
+            do {
+                let result = try JSONDecoder().decode(Film.self, from: data)
+                completion(result)
+            } catch {
+                print(error)
+            }
+        }
+        task.resume()
+    }
+
+    func loadVideos(index: Int, completion: @escaping ([VideoId]) -> Void) {
+        var components = URLComponents()
+        components.scheme = Constants.componentScheme
+        components.host = Constants.componentsHost
+        components.path = Constants.componentsPath + "\(index)" + Constants.videos
+        components.queryItems = [queryItemKey, queryItemLanguage]
+
+        guard let url = components.url else { return }
+        print(url)
+        let task = session.dataTask(with: url) { data, _, _ in
+            guard let data = data else { return }
+            do {
+                let result = try JSONDecoder().decode(ResultVideos.self, from: data)
+                completion(result.results)
+            } catch {
+                print(error)
+            }
+        }
+        task.resume()
+    }
```

Comment:
```suggestion
        var components = URLComponents()
        components.scheme = Constants.componentScheme
        components.host = Constants.componentsHost
        components.path = Constants.componentsPath + "\(index)" + Constants.videos
        components.queryItems = [queryItemKey, queryItemLanguage]

        guard let url = components.url else { return }
        print(url)
        let task = session.dataTask(with: url) { data, _, _ in
            guard let data = data else { return }
            do {
                let result = try JSONDecoder().decode(ResultVideos.self, from: data)
                completion(result.results)
            } catch {
                print(error)
            }
        }
        task.resume()
    }
```

повторяется одно и тоже 3 раза, по хорошему надо было сделать через дженерики